### PR TITLE
Rename and wrap ErasedTimer

### LINKED ISSUE #2143: Rename `ErasedX` to `AnyX`
We already did this for GPIO in #2128, but we should also do this for the other APIs to be consistent - I think this is just the timers right now. Not only does it make more sense IMO, it's also more consistent with the rest of the ecosystem (embassy hals).

### PR DESCRIPTION
**Pull request by bugadani**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

### Pull Request Details 📖

Stripped out of #2136 as at least this part is uncontroversial.

Closes #2143

### CODE REVIEW

File: esp-hal/MIGRATING-0.20.md [hunk 1]
```diff
@@ -74,15 +74,6 @@ let mut wakeup_pins: [&mut dyn RtcPin; 2] = [
 let ext1 = Ext1WakeupSource::new(&mut wakeup_pins, WakeupLevel::High);
 ```
 
-## `esp_hal::time::current_time` rename
-
-To avoid confusion with the `Rtc::current_time` wall clock time APIs, we've renamed `esp_hal::time::current_time` to `esp_hal::time::now()`.
-
-```diff
-- use esp_hal::time::current_time;
```
Comment (bugadani) [inline esp-hal/MIGRATING-0.20.md:82]: I found this diff trivial enough to not keep.

**Comment by bugadani** [comment]
Added `skip-changelog` due to esp-hal-embassy/esp-wifi changes

File: esp-hal-embassy/src/lib.rs [hunk 2]
```diff
@@ -69,24 +69,24 @@ pub trait TimerCollection {
 
 /// Helper trait to reduce boilerplate.
 ///
-/// We can't blanket-implement for `Into<ErasedTimer>` because of possible
+/// We can't blanket-implement for `Into<AnyTimer>` because of possible
 /// conflicting implementations.
-trait IntoErasedTimer: Into<ErasedTimer> {}
+trait IntoErasedTimer: Into<AnyTimer> {}
```
Comment (MabezDev) [inline esp-hal-embassy/src/lib.rs:74]: I suppose we should rename this `Into` trait too, just for consistency

**Review by MabezDev — CHANGES_REQUESTED** [review]

File: esp-hal-embassy/src/lib.rs (hunk 2 above)
Comment (bugadani) [reply esp-hal-embassy/src/lib.rs:74]: fair

**Review by MabezDev — APPROVED** [review]

**Pull request merged**